jaxrs/WFLY-15495_gRPC-support.adoc

```diff
@@ -0,0 +1,281 @@
+= Add support for gRPC
+:author:            Harald Pehl
+:email:             hpehl@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.redhat.com/browse
+
+== Overview
+
+The goal of this RFE is to provide support for gRPC in WildFly. https://grpc.io/[gRPC] is a high-performance RPC framework.
+It can efficiently connect services implemented using various languages and frameworks.
+It is also applicable in the last mile of distributed computing to connect devices, mobile applications, and browsers to backend services.
+
+This RFE will add a gRPC subsystem to WildFly.
+It:
+
+* supports implementing gRPC services
+* supports consuming gRPC services
+* integrates with other WildFly components such as CDI, MP health, security or transaction
+* allows plain-text communication as well as TLS, and TLS with mutual authentication
+
+The gRPC subsystem provides a default configuration for all gRPC services and clients.
+The major part of the gRPC subsystem is a deployment unit processor which scans deployments for gRPC services, clients and interceptos.
+
+=== Jakarta RPC Specification
+
+There's a Jakarta EE proposal for a specification called https://projects.eclipse.org/reviews/jakarta-rpc-creation-review[Jakarta RPC].
+This specification aims to make gRPC services and clients easier to implement in Java.
+The specification proposes a programming model for gRPC similar to JAX-RS and MicroProfile REST Client.
+As the time of writing, the proposal is not an official specification yet and very much wip.
+
+The scope of this RFE is *not* to implement Jakarta RPC.
+If it becomes a Jakarta EE specification at some point in time, there might be a follow-up RFE to support such a specification.
+Instead, this RFE is about to bring gRPC support to WildFly.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/WFLY-15495[WFLY-15495]
+* {issue-base-url}/EAP7-1819[EAP7-1819]
+
+=== Related Issues
+
+* {issue-base-url}/WFLY-13530[WFLY-13530]
+* {issue-base-url}/EAP7-1536[EAP7-1536]
+
+=== Dev Contacts
+
+* mailto:hpehl@redhat.com[Harald Pehl]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+== Requirements
+
+=== Hard Requirements
+
+* The gRPC subsystem provides a default configuration
```

Comment:
Does it make sense to mention the Galleon layer you plan to create/use? And the dependencies?

Reply:
Right now the [layer](https://github.com/hpehl/wildfly-grpc-feature-pack/blob/main/feature-pack/src/main/resources/layers/standalone/grpc/layer-spec.xml) for gRPC is pretty straightforward. It just has one dependency on CDI. This might change during development, though. Don't know if it makes sense to define the layer here, if it might change during development. WDYT?

Reply:
I've added a section about the Galleon layer.